### PR TITLE
feat(i18n): add Burmese(my) translation

### PR DESCRIPTION
**Pull request by myonaingwinn**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

---
This pull request aims to add Burmese language support as referenced in [issue #17764](https://github.com/element-plus/element-plus/issues/17764).
Once the pull request is ready, could you please review the code and approve it? Thank you!

### CODE REVIEW

File: packages/locale/lang/my.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        previous: 'ရှေ့သို့',
+        finish: 'ပြီးပြီ',
+      },
+      tree: {
+        emptyText: 'ဒေတာမရှိပါ',
+      },
+      transfer: {
+        noMatch: 'ကိုက်ညီသောဒေတာမရှိပါ',
+        noData: 'ဒေတာမရှိပါ',
+        titles: ['စာရင်း ၁', 'စာရင်း ၂'], // to be translated
```
Comment (plainheart) [inline packages/locale/lang/my.ts:161]: Please remove the suffix comment if the corresponding item is translated.

**Review by plainheart — CHANGES_REQUESTED** [review]

**Comment by myonaingwinn** [comment]
@plainheart Thank you for your review! I've removed the suffix comments from the translated items as you suggested. Feel free to adjust it further if needed!

**Review by plainheart — APPROVED** [review]

File: packages/locale/lang/my.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
+export default {
+  name: 'my',
+  el: {
+    breadcrumb: {
+      label: 'Breadcrumb', // to be translated
```
Comment (btea) [inline packages/locale/lang/my.ts:5]: Is there a missing translation here?

Comment (myonaingwinn) [reply packages/locale/lang/my.ts]: @btea  Apologies for the late reply! I’ve fixed that. Please feel free to make any additional adjustments as needed. Let me know if you need any further modifications!

**Review by btea — APPROVED** [review]
Thanks!

File: packages/locale/lang/my.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      loading: 'ဝန်တင်နေသည်',
+      noMatch: 'ကိုက်ညီသောဒေတာမရှိပါ',
+      noData: 'ဒေတာမရှိပါ',
+      placeholder: 'ရွေးပါ',
+    },
+    mention: {
+      loading: 'ဝန်တင်နေသည်',
+    },
+    dropdown: {
+      toggleDropdown: 'Dropdown ဖွင့်/ပိတ်',
```
Comment (btea) [inline packages/locale/lang/my.ts:99]: Do the English words here need to be translated?

File: packages/locale/lang/my.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    },
+    messagebox: {
+      title: 'မက်ဆေ့ချ်',
+      confirm: 'အိုကေ',
+      cancel: 'ပယ်ဖျက်',
+      error: 'တရားမဝင်ထည့်သွင်းမှု',
+      close: 'ဤဒိုင်ယာလော့ဂ်ကို ပိတ်ပါ',
+    },
+    upload: {
+      deleteTip: 'ဖယ်ရှားရန် delete ကိုနှိပ်ပါ',
```
Comment (btea) [inline packages/locale/lang/my.ts:135]: Does `delete` need to be translated?

File: packages/locale/lang/my.ts (hunk 3 above)
Comment (myonaingwinn) [reply packages/locale/lang/my.ts]: @btea There isn't a direct translation for "Dropdown" in Burmese, but I’ve updated the translation to make it easier to understand.

File: packages/locale/lang/my.ts (hunk 4 above)
Comment (myonaingwinn) [reply packages/locale/lang/my.ts]: @btea Thank you for your very quick and detailed reviews! I’ve also updated this translation. Let me know if you need any more adjustments!

**Pull request merged**